app/auth/views.py

```diff
@@ -0,0 +1,343 @@
+                    return make_response(jsonify(response))
+            else:
+                response = {
+                    'message': 'Invalid email or password. Try again please.'
+                }
+                return make_response(jsonify(response)), 401
+        except Exception as e:
+            response = {
+                'message': str(e)
+            }
+            return make_response(jsonify(response)), 500
+
+
+class BucketlistView(MethodView):
+    """Handles bucketlist creation and manipulation"""
+
+    def post(self):
+        """Handles POST request for a new bucketlist"""
+        try:
+            access_token = request.headers.get('Authorization')
+            if access_token:
+                user_id = User.decode_token(access_token)
+                if isinstance(user_id, int):
+                    name = request.data.get('name', '')
+                    bucketlist = BucketList(name=name, created_by=user_id)
+                    bucketlist.save()
+                    response = jsonify({
+                        'id': bucketlist.id,
+                        'name': bucketlist.name,
+                        'date_created': bucketlist.date_created,
+                        'date_modified': bucketlist.date_modified,
+                        'created_by': user_id
+                    })
+
+                    return make_response(response), 201
+        except Exception as e:
+            response = {
+                'message': str(e)
+            }
+            return make_response(jsonify(response)), 500
+
+    def delete(self, id, **kwargs):
+        """Handles DELETE request to delete a bucketlist"""
+        try:
+            access_token = request.headers.get('Authorization')
+            if access_token:
+                user_id = User.decode_token(access_token)
+                if isinstance(user_id, int):
+                    bucketlist = BucketList.query.filter_by(id=id).first()
+                    if not bucketlist:
+                        abort(404)
+                    else:
+                        bucketlist.delete()
+                    return{"message":
+                           "bucketlist {} successfully deleted".format(
+                               bucketlist.id)}, 200
+                    # raise HTTP 404 response if id not found
+        except Exception as e:
+            response = {
+                'message': str(e)
+            }
+            return make_response(jsonify(response)), 500
+
+    def put(self, id, **kwargs):
+        """Handles the PUT request to edit bucketlist"""
+        try:
+            access_token = request.headers.get('Authorization')
+            if access_token:
+                user_id = User.decode_token(access_token)
+                if isinstance(user_id, int):
+                    bucketlist = BucketList.query.filter_by(id=id).first()
+                    if bucketlist:
+                        name = str(request.data.get('name', ''))
```

Comment:
If there's no name in the request data, we'll overwrite the preexisting bucketlist name with an empty string. This should probably default to the name of the bucketlist, e.g
```python
name = str(request.data.get('name', bucketlist.name))
```